astroid/inference.py

```diff
@@ -1139,10 +1141,12 @@ def infer_empty_node(
 nodes.EmptyNode._infer = infer_empty_node  # type: ignore[assignment]
 
 
-def _populate_context_lookup(call, context):
+def _populate_context_lookup(
+    call: nodes.Call, context: InferenceContext | None
+) -> dict[InferenceResult, InferenceContext]:
```

Comment:
Have you actually checked that this is `InferenceResult` and not `SuccesfulInferenceResult`?